ports/stm32/Makefile

```diff
@@ -534,6 +534,7 @@ endif
 
 endif
 
+OBJ += $(addprefix $(BUILD)/, $(SRC_O))
```

Comment:
A comment linking the bug and the workaround is needed to prevent anyone from rearranging this in the future.

Reply:
`HAL_SRC_C` also potentially contains weak symbols, so wouldn't it need to be moved before this too?

Reply:
@dlech you're very right about the comment, will add.

Actually the issue seems like it only affects weak symbols defined in assembler (probably due to a symbol flag or section name?), which is mentioned in this bug -  https://gcc.gnu.org/bugzilla/show_bug.cgi?id=83967 - that I didn't link and will link.

This is the kind of detail that will be useful to put in the comment. :grin:

Reply:
@dlech Done, PTAL and thanks for pointing that out!

Reply:
LGTM